doc/arch/adr-008-items-resource.md

```diff
@@ -0,0 +1,81 @@
+# Decision record: Items resource implementation
+
+## Context
+
+The [Items resource](https://openregister.github.io/specification/#items-resource)
+imposes an implementation challenge regarding how to order them in a
+pagination scenario. This ADR defines a way to create a deterministic order
+for this implementation.
+
+## Decision
+
+The Items set has no natural order. Items are identified by their hash. A way
+to define an order is to assign an index or timestamp based on order of
+arrival to the database.
+
+```sql
+CREATE TABLE IF NOT EXISTS item (
+  n serial,
+  sha256hex VARCHAR PRIMARY KEY,
+  content JSONB
+);
+
+CREATE INDEX item_idx ON item USING btree (n);
+
+SELECT sha256hex, content
+FROM item
+WHERE n > ?cursor
+ORDER BY n ASC
+FETCH NEXT ?size ROWS ONLY;
+```
+
+Where `?cursor` and `?size` are defined by the request. Defaults are `0` and
+`100` respectively.
+
+The notion of "cursor" is to be clear it is an obscure pointer rather than an
+actual page. This is commonly known as _seek method_ or _keyset pagination_.
+
+The result set should not suface the `n` field except as the next `cursor`
+(see next section).
+
+### Changes to the API
+
+Implement the `/items` endpoint as defined in the specification. This should
+look like:
+
+```http
+GET /items HTTP/1.1
+Host: country.register.gov.uk
+Accept: application/json
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Link: <?cursor=3>; rel="next"
```

Comment:
Should the API also support a link to the previous page?

Reply:
Good question, my understanding of the potential usage for this endpoint is that it doesn't require a previous link. So it would be a sort-of iterator that goes forward. If we find evidence that we need the previous page, we can reconsider.

Reply:
Ok cool. I agree it's better to leave out unless we have a strong use case.